pos-frontend/src/components/SignUp.jsx

```diff
@@ -0,0 +1,223 @@
+import { Container, Grid, CssBaseline, Typography, TextField, Box, Avatar, Button, Snackbar, Alert } from '@mui/material'
+import { Link, useNavigate } from "react-router-dom"
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined'
+import { useForm } from "react-hook-form"
+import axios from "axios"
+import { useState } from 'react'
+
+const initialState = {
+  open: false,
+  status: '',
+  message: ''
+}
+
+const SignUp = (props) => {
+
+  const { register , handleSubmit, formState: { errors }, watch, getValues, reset, setFocus, setError } = useForm();
+  const [ message, setMessage ] = useState(initialState);
+  const navigate = useNavigate()
+
+  const onSubmit = () => {
+    const userData = getValues();
+
+    axios.post("http://localhost:5000/api/auth/signup", userData).then((res) => {
+
+      console.log(res);
+
+      if(res.data.status === 'error') {
+
+        if(res.status == 208) {
+          setError('email', { message: 'Email is already registered. Please try to register with another email.' })
+          setFocus('email');
+        }
+        setMessage({
+          open: true,
+          status: 'error',
+          message: res.data.message
+        })
+      }
+      else {
+        setMessage({
+          open: true,
+          status: 'ok',
+          message: 'User is registred successfully!'
+        });
+        reset();
+        navigate("/");
+      }
+
+    })
+    .catch((err) => console.log(err));
+  }
+
+  const handleClose = () => {
+    setMessage(initialState);
+  }
+
+  return (
+    <>
+      <Box component="main" sx={{ display: 'flex', height:'100vh', alignItems: 'center', justifyContent: 'center'}}>
+        <CssBaseline />
+
+        <Snackbar 
+          anchorOrigin={{ vertical: 'top', horizontal: 'center'}}
+          onClose={handleClose}
+          open={message.open}
+          key={"top"}
+          message={message.message}
+          autoHideDuration={6000} >
+
+            <Alert onClose={handleClose} severity={message.status === 'ok' ? "success" : "error" } sx={{ width: '100%' }}>
+              {message.message}
+            </Alert>
+
+        </Snackbar>
+
+
+        <Grid container component="div" sx={{ height: '100vh', width: "100%" }}>
+          <Grid 
+            item 
+            xs={false} 
+            sm={6} 
+            sx={{
+              backgroundImage: "url(/images/login-img1.jpg)",
```

Comment:
make a ts file file from where you can export all images

Reply:
Done